keanu-project/src/main/java/io/improbable/keanu/vertices/generic/probabilistic/discrete/CategoricalVertex.java

```diff
@@ -23,6 +25,27 @@
         return new CategoricalVertex<>(defensiveCopy(selectableValues));
     }
 
+    public static <T> CategoricalVertex<T> of(DirichletVertex vertex, List<T> categories) {
+        final int length = Math.toIntExact(TensorShape.getLength(vertex.getShape()));
```

Comment:
I still maintain that this can just be a long - you don't need to cast at all:

        final long length = TensorShape.getLength(vertex.getShape());

Reply:
I'll get this changed up